crates/orchestrator/src/discovery/monitor.rs

```diff
@@ -1,59 +1,90 @@
+use crate::discovery::location_service::LocationService;
```

Comment:
the DiscoveryMonitor was refactored to have 2 internal components: `NodeFetcher` and `Updater`. `NodeFetcher` fetches the nodes from the DHT and returns them, while `Updater` performs the same "syncing" logic and updating the node store as previously.